packages/edit-navigation/src/components/header/index.js

```diff
@@ -64,50 +70,78 @@ export default function Header( {
 					<DropdownMenu
 						icon={ null }
 						toggleProps={ {
+							children: __( 'Switch menu' ),
+							'aria-label': __(
+								'Switch menu, or create a new menu'
+							),
 							showTooltip: false,
-							children: __( 'Select menu' ),
 							isTertiary: true,
 							disabled: ! menus?.length,
 							__experimentalIsFocusable: true,
 						} }
 						popoverProps={ {
+							className:
+								'edit-navigation-header__menu-selection-dropdown',
 							position: 'bottom left',
 						} }
 					>
-						{ () => (
-							<MenuGroup>
-								<MenuItemsChoice
-									value={ selectedMenuId }
-									onSelect={ onSelectMenu }
-									choices={ menus.map( ( menu ) => ( {
-										value: menu.id,
-										label: menu.name,
-									} ) ) }
-								/>
-							</MenuGroup>
+						{ ( { onClose } ) => (
```

Comment:
I think the code would be easier to read if we could separate this function to another file, or if we define elements props outside of JSX

Reply:
I think we also want to re-use this for your other PR (https://github.com/WordPress/gutenberg/pull/29201).

I've moved the menu items into a component `MenuSwitcher` in ac70e42